### ISSUE TITLE
output for time series tasks

### ISSUE THREAD

**Issue by lanxiang1017** [locuslab/TCN#50]
Hi Jerry, thanks for the nice work, I'm trying to use tcn for a multi-variate time series task, suppose two data points with different sequence length are x1 = [t1, t2, t3, t4], x2 = [t1, t2, t3, t4, t5, t6], and corresponding labels for each time-step are [y1, y2, y3, y4] and [y1, y2, y3, y4, y5, y6], padding them to the same length: x1 = [0, 0, t1, t2, t3, t4], x2 = [t1, t2, t3, t4, t5, t6] for mini-batch input.

My question is, given x1 and x2, how the model output the prediction for each time-step? like for x1 we want the model output [z0, z1, z2, z3, z4, z5] and we only take [z2, z3, z4, z5] for binary cross entropy loss with it's true label [y1, y2, y3, y4].

my current approach is refer to the mnist example, by modify the models forward with:

>    def forward(self, inputs):
        """Inputs dimension (N, C_in, L_in)"""
        seq_len = inputs.shape[2]
        y1 = self.tcn(inputs)
        o = torch.cat([self.linear(y1[:,:,i]) for i in range(0, seq_len)], dim=1)
        return o

this will output the same length as input, but I'm not sure if this is the correct understanding for tcn model, or there should be some other approaches. 

Thank you.

**Comment by jerrybai1995** [comment]
For MNIST you only need one label prediction, so I only took the last time step for output. But generally, as is in your case, I think you are doing it in the correct way to take [z2, z3, z4, z5] to compute the loss with the groundtruth [y1, y2, y3, y4].

**Comment by lanxiang1017** [comment]
yes, and in order to output for each time-step, are these two approaches the same?
`y1 = self.tcn(inputs) `
`o = self.linear(y1.transpose(1, 2))  # refer to copy memory task`
and
`y1 = self.tcn(inputs)`
`o = torch.cat([self.linear(y1[:,:,i]) for i in range(0, seq_len)], dim=1)`

**Comment by jerrybai1995** [comment]
They should be, except that I think the second one is slower in practice because you are not exploiting the parallelism.
